pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -212,15 +212,19 @@ func (rc *reconciler) reconcile(ctx context.Context) {
 
 			isHealthy, err := rc.nodeIsHealthy(attachedVolume.NodeName)
 			if err != nil {
-				logger.Error(err, "Failed to get health of node", "node", klog.KRef("", string(attachedVolume.NodeName)))
+				logger.V(5).Info("Failed to get health of node",
```

Comment:
An error should always be logged. Changing this to V5 means we may miss this when we need to know an error has occurred.

Reply:
Hi @xing-yang thanks for the prompt review.

The motivation for this PR is user reports of over 10 million logs in a short period of time from these two errors as a result of normal scale-down events. These get triggered for every single volume on the node, which results in an overwhelming amount of logs.

In this reconciler code, there are already errors which are logged at the `logger.V(5).Info` level - presumably to prevent flooding logs, here is an example: https://github.com/kubernetes/kubernetes/blob/3cc3b002ba13b2a9593d5b12edb8dbdc0eb727ac/pkg/controller/volume/attachdetach/reconciler/reconciler.go#L239

At the very least, given that out of service taints are optionally applied by users, is it wise to log at the `logger.Error` level?

As for logging node health check at the `logger.Error` level, if that is your recommendation then I support it.

Reply:
I agree that we should not log always here. We need to be very careful about what we log in reconciler.go because it runs every few ms without any throttling. It looks like this logic did not take into account the scenario of autoscaling down nodes, which is an error to expect in typical operations.

@xing-yang what is the behavior we want when we encounter an error trying to get node information? Should we 1) continue to try to detach or 2) not detach and allow the normal node deletion flow to proceed?

Reply:
Hi @xing-yang, could you shed some light on the above discussion?